eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2467,8 +2467,21 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
         val redeemableHtlcTxs = htlcTxs.values.flatten.map(tx => PublishFinalTx(tx, tx.fee, Some(commitTx.txid)))
         List(PublishFinalTx(commitTx, commitInput, "commit-tx", Closing.commitTxFee(commitments.commitInput, commitTx, isFunder), None)) ++ (claimMainDelayedOutputTx.map(tx => PublishFinalTx(tx, tx.fee, None)) ++ redeemableHtlcTxs ++ claimHtlcDelayedTxs.map(tx => PublishFinalTx(tx, tx.fee, None)))
       case _: Transactions.AnchorOutputsCommitmentFormat =>
-        val claimLocalAnchor = claimAnchorTxs.collect { case tx: Transactions.ClaimLocalAnchorOutputTx => PublishReplaceableTx(tx, commitments) }
-        val redeemableHtlcTxs = htlcTxs.values.collect { case Some(tx) => PublishReplaceableTx(tx, commitments) }
+        val redeemableHtlcTxs = htlcTxs.values.collect {
+          case Some(tx) =>
+            val htlc_opt = tx match {
+              case _: Transactions.HtlcSuccessTx => commitments.localCommit.spec.findIncomingHtlcById(tx.htlcId).map(_.add)
+              case _: Transactions.HtlcTimeoutTx => commitments.localCommit.spec.findOutgoingHtlcById(tx.htlcId).map(_.add)
+            }
+            val deadline = htlc_opt.map(_.cltvExpiry.toLong).getOrElse(nodeParams.currentBlockHeight + nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)
```

Comment:
`.getOrElse(nodeParams.currentBlockHeight + nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)` looks suspicious: if `htlc_opt` is undefined it's a bug?

Reply:
Yes, IIRC this should always be defined, but it's hard to express at the type level so I used a fallback.
I'll see if I can refactor that to ensure we always have a value here.

Reply:
For example:
```scala
val redeemableHtlcTxs = htlcTxs.values.collect {
  case Some(tx: Transactions.HtlcSuccessTx) => (tx, commitments.localCommit.spec.findIncomingHtlcById(tx.htlcId).map(_.add))
  case Some(tx: Transactions.HtlcTimeoutTx) => (tx, commitments.localCommit.spec.findOutgoingHtlcById(tx.htlcId).map(_.add))
}
  .collect {
    case (tx, Some(htlc)) =>
      val deadline = htlc.cltvExpiry.toLong
      PublishReplaceableTx(tx, commitments, deadline)
  }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/b0667538d8dbf95c15b4c77c042787ba9930ea59